api/example.com/resource/gpu/v1alpha1/validate_test.go

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2025 The Kubernetes Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGpuConfigValidate(t *testing.T) {
+	tests := map[string]struct {
+		gpuConfig *GpuConfig
+		expected  string
+	}{
+		"empty GpuConfig": {
+			gpuConfig: &GpuConfig{},
+			expected:  "no sharing strategy set",
+		},
+		"empty GpuConfig.Sharing": {
+			gpuConfig: &GpuConfig{
+				Sharing: &GpuSharing{},
+			},
+			expected: "unknown GPU sharing strategy: ",
+		},
```

Comment:
I'd also add this test case:
```
		"unknown GPU sharing strategy": {
			gpuConfig: &GpuConfig{
				Sharing: &GpuSharing{
					Strategy: "unknown",
				},
			},
			expected: errors.New("unknown GPU sharing strategy: unknown"),
		},
```
BTW, this test case shows that[ this code](https://github.com/kubernetes-sigs/dra-example-driver/blob/main/api/example.com/resource/gpu/v1alpha1/validate.go#L65) is unreachable.